src/Infrastructure/Controller/SingleSignOnController.php

```diff
@@ -20,17 +20,94 @@
 
 namespace StepupReadId\Infrastructure\Controller;
 
+use Psr\Log\LoggerInterface;
+use StepupReadId\Application\ReadySession\GetReadySessionQuery;
+use StepupReadId\Domain\ReadySession\Model\ReadySession;
+use StepupReadId\Infrastructure\Services\ReadySession\ReadySessionStateHandlerInterface;
+use Surfnet\GsspBundle\Saml\ResponseContextInterface;
+use Surfnet\GsspBundle\Service\StateHandlerInterface;
+use Surfnet\SamlBundle\SAML2\AuthnRequest;
+use Surfnet\SamlBundle\SAML2\ReceivedAuthnRequest;
 use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\HttpFoundation\Request;
 use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Messenger\HandleTrait;
+use Symfony\Component\Messenger\MessageBusInterface;
 use Symfony\Component\Routing\Annotation\Route;
+use function sprintf;
 
+/**
+ * @Route("/saml/sso", name="readid_saml_sso", methods={"GET"})
+ * @SuppressWarnings(PHPMD.CouplingBetweenObjects)
+ */
 final class SingleSignOnController extends AbstractController
```

Comment:
Please consider offloading some of the logic from this Controller to a service. This way you will probably decrease coupling in the controller. And this will make the code more easily testable.

You might want to do this at a later stage when all interactions are finalized and a more clear set of services can be extracted.

Reply:
My bad. I forgot to remove this line. I move the AuthnRequest extraction from the Http Request to an ArgumentResolver just to resolve that.